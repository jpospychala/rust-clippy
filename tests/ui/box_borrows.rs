#![warn(clippy::all)]
#![allow(clippy::boxed_local, clippy::needless_pass_by_value)]
#![allow(clippy::blacklisted_name)]

pub struct MyStruct {}

pub struct SubT<T> {
    foo: T,
}

pub enum MyEnum {
    One,
    Two,
}

pub fn test<T>(foo: Box<&T>) {}

pub fn test1(foo: Box<&usize>) {}

pub fn test2(foo: Box<&MyStruct>) {}

pub fn test3(foo: Box<&MyEnum>) {}

pub fn test4(foo: Box<&&MyEnum>) {}

pub fn test5(foo: Box<Box<&usize>>) {}

pub fn test6(foo: Box<SubT<&usize>>) {}

fn main() {}
