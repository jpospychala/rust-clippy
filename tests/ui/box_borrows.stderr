error: usage of `Box<&T>`
  --> $DIR/box_borrows.rs:16:21
   |
LL | pub fn test<T>(foo: Box<&T>) {}
   |                     ^^^^^^^
   |
   = note: `-D clippy::box-borrows` implied by `-D warnings`
   = help: try `&T`

error: usage of `Box<&T>`
  --> $DIR/box_borrows.rs:18:19
   |
LL | pub fn test1(foo: Box<&usize>) {}
   |                   ^^^^^^^^^^^
   |
   = help: try `&usize`

error: usage of `Box<&T>`
  --> $DIR/box_borrows.rs:20:19
   |
LL | pub fn test2(foo: Box<&MyStruct>) {}
   |                   ^^^^^^^^^^^^^^
   |
   = help: try `&MyStruct`

error: usage of `Box<&T>`
  --> $DIR/box_borrows.rs:22:19
   |
LL | pub fn test3(foo: Box<&MyEnum>) {}
   |                   ^^^^^^^^^^^^
   |
   = help: try `&MyEnum`

error: usage of `Box<&T>`
  --> $DIR/box_borrows.rs:24:19
   |
LL | pub fn test4(foo: Box<&&MyEnum>) {}
   |                   ^^^^^^^^^^^^^
   |
   = help: try `&&MyEnum`

error: usage of `Box<&T>`
  --> $DIR/box_borrows.rs:26:23
   |
LL | pub fn test5(foo: Box<Box<&usize>>) {}
   |                       ^^^^^^^^^^^
   |
   = help: try `&usize`

error: aborting due to 6 previous errors

